File: backend/src/main/java/com/ums/ums_backend/entity/Enrollment.java
package com.ums.ums_backend.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDate;

/**
 * Represents a student's enrollment in a course for a specific semester with grade and status tracking.
 */
@Entity
@Getter
@Setter
@Table(
        uniqueConstraints = {
                @UniqueConstraint(
                        columnNames = {"student_id", "course_id", "semester_id"}
                )
        }
)
public class Enrollment {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "semester_id", nullable = false)
    private Semester semester;

    @Column(nullable = false)
    private LocalDate enrollmentDate;

    @ManyToOne
    @JoinColumn(name = "student_id", nullable = false)
    private Student student;

    @ManyToOne
    @JoinColumn(name = "course_id", nullable = false)
    private Course course;

    private Double grade;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private EnrollmentStatus enrollmentStatus;

}

File: frontend/src/app/features/enrollment/models/enrollment.ts
import {EnrollmentStatus} from '../../../core/enums/EnrollmentStatus';
import {StudentSummary} from '../../student/models/StudentSummary';
import {CourseSummary} from '../../course/models/CourseSummary';
import {Semester} from '../../semester/models/semester';

export interface Enrollment {

  id: number,
  semester: Semester,
  enrollmentDate: string,
  student: StudentSummary,
  course: CourseSummary,
  grade: number,
  enrollmentStatus: EnrollmentStatus

}

export interface EnrollmentCreateRequest {

  studentId: number,
  courseId: number,
  semesterId: number,
  enrollmentDate: string,
  enrollmentStatus: EnrollmentStatus
  grade?: number,

}
